pyccel/codegen/printing/ccode.py

```diff
@@ -208,8 +208,37 @@ def _print_PyccelNot(self, expr):
         a = self._print(expr.args[0])
         return '!{}'.format(a)
 
+    def _print_PyccelMod(self, expr):
+        self._additional_imports.add("math.h")
+
+        first = self._print(expr.args[0])
+        second = self._print(expr.args[1])
+
+        if expr.dtype is NativeInteger():
+            return "{} % {}".format(first, second)
+        
+        if expr.args[0].dtype is NativeInteger():
+            first = self._print(PythonFloat(expr.args[0]))
+        if expr.args[1].dtype is NativeInteger():
+            second = self._print(PythonFloat(expr.args[1]))
+        return "fmod({}, {})".format(first, second)
+
+    def _print_PyccelPow(self, expr):
+        self._additional_imports.add("math.h")
+        if expr.args[0].dtype is NativeInteger():
+            base = self._print(PythonFloat(expr.args[0]))
+        else:
+            base = self._print(expr.args[0])
+        if expr.args[1].dtype is NativeInteger():
+            e = self._print(PythonFloat(expr.args[1]))
+        else:
+            e = self._print(expr.args[1])
+        if expr.dtype is NativeInteger():
+            return '(long)pow({}, {})'.format(base, e)
```

Comment:
Please use cast functions to cast

Reply:
Sorry, I've just realised that it is difficult (impossible?) to use the casting function here as it is around the entire object